File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:

    int max = 0;

    int maxDepth(TreeNode* root) {
        inorder(root,0);
        return max;
    }

    void inorder(TreeNode* root, int depth){
        if(root==nullptr){
            if(depth>max){
                max = depth;
            }
            return;
        }
        depth++;
        inorder(root->left,depth);
        inorder(root->right,depth);
    }
};

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int max;
    public int maxDepth(TreeNode root) {
        if(root==null){
            return 0;
        }
        int lvl = 0;
        print(root,lvl);
        return max;
    }
    public void print(TreeNode root, int lvl){
        lvl++;
        max = Math.max(max,lvl);
        if(root.left!=null) print(root.left,lvl);
        if(root.right!=null) print(root.right,lvl);
    }
}